mne/encoding/feature.py

```diff
@@ -0,0 +1,269 @@
+"""Classes for transforming, preprocessing, and fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy.sparse import spmatrix, csr_matrix
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Creates delayed versions of a time series.
+
+    Parameters
+    ----------
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means time points in the past,
+        positive means time points in the future. Default is no delays.
+    sfreq: float
+        The sampling frequency of the series. Defaults to 1.0
+
+    Attributes
+    ----------
+    ``names_`` : ndarray, shape (n_features * n_delays, 2)
+        If fit, a collection of `(feature_name, delay)` pairs
+        corresponding to columns of the output of `transform`.
+    """
+    def __init__(self, delays=0., sfreq=1.):
+        delays, _, sfreq = _check_delayer_params(
+            delays, sfreq)
+        self.delays = delays
+        self.sfreq = sfreq
+
+    def fit(self, X, y=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+            Training data.
+        y : array
+            Target values.
+
+        Returns
+        -------
+        self
+        """
+        return self
+
+    def transform(self, X, y=None):
+        """Create a time-lagged representation of X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_times, n_features)
+            The input data to be time-lagged and returned.
+
+        Returns
+        -------
+        X_delayed : array, shape (n_times, n_features * n_delays)
+            The delayed features of X.
+        """
+        X_delayed = delay_time_series(X, delays=self.delays,
+                                      sfreq=self.sfreq)
```

Comment:
I figure it just gives more flexibility to do whatever the user
wants...I've found it restrictive when using functions that require you use
all possible lags when delaying a timeseries. (also just FYI I've got 3
deadlines in the next 2 weeks so I won't be able to make big changes to
this PR for a little bit...if I can get some of those deadlines out of the
way sooner than later then I will work on this)
-- 

On Mon, Aug 22, 2016 at 12:24 AM Alexandre Gramfort <
notifications@github.com> wrote:

> In mne/encoding/feature.py
> <https://github.com/mne-tools/mne-python/pull/3310#discussion_r75628317>:
>
> > +
> > +    def transform(self, X, y=None):
> > +        """Create a time-lagged representation of X.
> > +
> > +        Parameters
> > +        ----------
> > +        X : array-like, shape (n_times, n_features)
> > +            The input data to be time-lagged and returned.
> > +
> > +        Returns
> > +        -------
> > +        X_delayed : array, shape (n_times, n_features * n_delays)
> > +            The delayed features of X.
> > +        """
> > +        X_delayed = delay_time_series(X, delays=self.delays,
> > +                                      sfreq=self.sfreq)
>
> is there a reason why you downsample the delays rather than resample the
> input time series?
>
> fair enough for using roll and add Hankel to docstrings
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/3310/files/e9c40309c15a581b0901f5b0784aacca5d2e0e6c#r75628317>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABwSHRF8ddzmePfUa2L8UyB7S3zqkx-1ks5qiU7DgaJpZM4I28Ut>
> .
>